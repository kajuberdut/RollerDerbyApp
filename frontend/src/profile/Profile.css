
/* note that this doesnt seem to be working but inline styles were working... */
/* .PROFILE {
    background-image: none;
    width: 100%;
    display: flex;
    border: 30px solid blue;
    background-color: yellow;
    
}

.Profile {
    margin-right: 30%;
    background-color: green;
} */

/* .MDBCardText {
    word-wrap: break-word;
  } */