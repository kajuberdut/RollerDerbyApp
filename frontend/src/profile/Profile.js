import React, { useState, useEffect, useContext } from "react";
import FastApi from "../Api";
import { useParams} from "react-router-dom";
// import skateImg from "../public/logo512.png"
// import defaultImg from "./images/skater_02"



// ! are you going to make a distinction betweeen everyone elses profile and the users profile... and the answer is probably yes... so you may want to have a /profile and a /users/derbyName

import UserContext from "../multiUse/UserContext";
import { MDBCol, MDBContainer, MDBRow, MDBCard, MDBCardText, MDBCardBody, MDBCardImage, MDBBtn, MDBTypography } from 'mdb-react-ui-kit';


/**  
 * User detail form
 */

// ! issue I am not storing all th data in the user information so Its not being updated all the way

function Profile() {

   /** Get url handle and set jobs and is loading in state*/
    const username = useParams(); 
    const [isLoading, setIsLoading] = useState(true);
    // const { user } = useContext(UserContext);
    // const [user, setUser] = useState(); 
    const [rulesets, setRulesets] = useState();
    const [positions, setPositions] = useState();
    const [location, setLocation] = useState();
    const [insurances, setInsurances] = useState();
    const [city, setCity] = useState();
    const [state, setState] = useState();
    const [phoneNumber, setPhoneNumber] = useState();

    // retrieve user from localStorage so that page can be refreshed 
    const user = JSON.parse(localStorage.getItem('user'));
    console.log("user in profile.js", user)

    
    
    useEffect(() => {
      function fetchData() {
          // Fetch ruleset data
          if (user.ruleset) {
            getUserRulesets()
          }
          // Fetch position data
          if (user.position) {
            getUserPosition()
          }
          // Fetch location data
          if (user.locationId) {
            // console.log("user.location_id is running")
            getUserLocation()
          }
          if (user.insurance) {
            getUserInsurance()
          }

          if(user.phoneNumber){
            console.log("IF STATEMENT IS RUNNING")
            formatPhoneNumber()
          }
        }
  
      fetchData();
    }, []); 

  async function getUserRulesets() {
    let rsArr = []
    for(let rs of user.ruleset) {
      let ruleset = await FastApi.getRuleset(rs.rulesetId);
      rsArr.push(ruleset.name)
    }
    let userRulesets = rsArr.join(", ")
    setRulesets(userRulesets)
  }

  async function getUserPosition() {
    let posArr = []
    for(let pos of user.position) {
      let position = await FastApi.getPosition(pos.positionId);
      posArr.push(position.position)
    }
    let userPositions = posArr.join(", ")
    setPositions(userPositions)
  }

  async function getUserInsurance() {
    let insArr = []
    for(let ins of user.insurance) {
      let insurance = await FastApi.getInsurance(ins.insuranceId);
      insArr.push(insurance.type + ":")
      insArr.push(ins.insuranceNumber)
    }
    if(insArr.length > 2){
      insArr[1] += ","
    }    
    let userInsurances = insArr.join(" ")
    setInsurances(userInsurances)
  }


  async function getUserLocation() {
    let userLocation = await FastApi.getLocation(user.locationId);
    setCity(userLocation.city)
    setState(userLocation.state)
  }

  async function formatPhoneNumber() {
    let formPhoneNum = user.phoneNumber.slice(0, 3) + '-' + user.phoneNumber.slice(3, 6) + '-' + user.phoneNumber.slice(6);
    setPhoneNumber(formPhoneNum)
  }

    /** Render cards */
    // ! will need to rework this

    return (
      <div className="PROFILE" style={{backgroundColor: 'transparent', padding: '100px'}} >

        <MDBContainer>
          <MDBRow className="justify-content-center align-items-center h-100"> 
            <MDBCol lg="9" xl="10">
              <MDBCard style={{minWidth: '300px', minHeight: '700px', marginTop: '50px', boxShadow: '0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2)'}}>
                <div className="rounded-top text-white d-flex flex-row" style={{ backgroundColor: '#000', height: '300px'}}>

                  <div className="ms-4 mt-5 d-flex flex-column" style={{ width: '250px' }}>    

                  <MDBCardImage src="/skater_02.svg"
                      alt="Skater placeholder image" className="mt-4 mb-2 img-thumbnail" fluid style={{ width: '250px', height: '330px', zIndex: '1', backgroundColor: '#d1d2d4', border: '4px solid white', boxShadow: '0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2)'}} /> 
                  </div>
                  <a href="/setup/profile">
                    <button type="button" className="btn btn-outline-dark"  data-mdb-ripple-color="dark"
                      style={{zIndex: 1, height: '40px', backgroundColor: '#d1d2d4', position: 'absolute', right: '20px', marginTop: '10px', fontSize: '15px'}}>
                      Edit
                    </button>
                  </a>
                  <div className="ms-3" style={{ marginTop: '200px'}}>
                    <MDBTypography tag="h4">{user.username} #{user.primaryNumber}</MDBTypography>
                    {/* {user.first_name && user.last_name && <MDBCardText>{user.first_name} {user.last_name}</MDBCardText>} */}
                    {user.locationId && city && state && <MDBCardText>{city}, {state}</MDBCardText>}
                  </div>
                </div>
                <div className="p-4 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                  <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                    { user.level && <div>
                      <MDBCardText className="mb-1 h5" style={{marginRight: '30px'}}>{user.level}</MDBCardText>
                      <MDBCardText className="small text-muted mb-0"style={{marginRight: '30px'}}>level</MDBCardText>
                    </div>
                    }
                    { user.position && <div>
                      <MDBCardText className="mb-1 h6">{positions}</MDBCardText>
                      <MDBCardText className="small text-muted mb-0" style={{marginRight: '30px', marginTop: '7px'}}>positions</MDBCardText>
                    </div>
                    }
                    { user.ruleset &&
                    <div className="px-3">
                      <MDBCardText className="mb-1 h6" style={{marginLeft: '30px'}}>{rulesets}</MDBCardText>
                      <MDBCardText className="small text-muted mb-0" style={{marginLeght: '30px', marginTop: '7px'}}>known rulesets</MDBCardText>
                    </div>
                    }
                  </div>
                </div>
                <MDBCardBody className="text-black p-4">

                { user.about && <div className="mb-5">
                    <p className="lead fw-normal mb-1">About</p>
                    <div className="p-4" style={{ backgroundColor: '#f8f9fa' }}>
                      <MDBCardText className="font-italic mb-1">{user.about}</MDBCardText>
                    </div>
                  </div>
                  }
                  { user.associatedLeagues && <div className="mb-5">
                    <p className="lead fw-normal mb-1">Associated Leagues</p>
                    <div className="p-4" style={{ backgroundColor: '#f8f9fa' }}>
                      <MDBCardText className="font-italic mb-1">{user.associatedLeagues}</MDBCardText>
                    </div>
                  </div>
                  }
                  { user.facebookName && <div className="mb-5">
                    <p className="lead fw-normal mb-1">You can find me on facebook: {user.facebookName}</p>
                  </div>
                  }
                </MDBCardBody>


                <MDBContainer style={{paddingBottom: '100px'}}>
                          <MDBRow className="justify-content-center align-items-center h-100"> 
                            <MDBCol lg="9" xl="10">
                              <MDBCard style={{minWidth: '300px', minHeight: '700px', marginTop: '50px', boxShadow: '0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2)'}}>
                                <div>
                                  <h2>Private Information </h2>
                                  <h3>Will be shared with team admin only.</h3>
                                  <div className="ms-4 mt-5 d-flex flex-column" style={{ width: '250px' }}>    
                                      <a href="/setup/private">
                                          <button type="button" className="btn btn-outline-dark"  data-mdb-ripple-color="dark"
                                            style={{zIndex: 1, height: '35px', backgroundColor: '#d1d2d4', position: 'absolute', top: '10px', right: '20px', marginTop: '10px', fontSize: '15px'}}>
                                            Edit
                                          </button>
                                      </a>
                                  </div>
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style= {{marginTop: '2px'}}>
                                    </div>
                                </div>

                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Name:
                                    </div>
                                    {user.firstName && user.lastName && <MDBCardText style={{marginLeft: '150px', marginTop: '5px'}} tag="h4">{user.firstName} {user.lastName}</MDBCardText>}
                                    </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>

                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Email:
                                    </div>
                                    <MDBCardText style={{marginLeft: '146px', marginTop: '5px'}} tag="h4">{user.email} </MDBCardText>
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>

                                
                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Phone #:
                                    </div>
                                    {user.phoneNumber && <MDBTypography style={{marginLeft: '126px', marginTop: '5px'}} tag="h4">{phoneNumber}</MDBTypography>}
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>

                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Insurances:
                                    </div>

                                    { insurances &&
                                          <MDBTypography style={{marginLeft: '100px', marginTop: '5px'}} tag="h4">{insurances}</MDBTypography>
                                    }
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>

                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Additional Info:
                                    </div>

                                    {user.additionalInfo && 
                                        <MDBTypography style={{marginLeft: '56px', marginTop: '5px'}} tag="h4">{user.additionalInfo}</MDBTypography>
                                    }
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>

                                <div className="ms-3 d-flex" style={{ marginTop: '0px'}}>
                                    <div className="m-0 text-start" style={{fontSize: '24px', fontFamily: 'initial', fontWeight: 'bold'}}>Secondary #:
                                    </div>

                                    {user.secondaryNumber != null && 
                                        <MDBTypography style={{marginLeft: '90px', marginTop: '5px'}} tag="h4">{user.secondaryNumber}</MDBTypography>
                                    }
                                </div>

                                <div className="p-2 text-black" style={{ backgroundColor: '#f8f9fa' }}>
                                    <div className="d-flex justify-content-end text-center py-1" style={{marginTop: '2px'}}>
                                    </div>
                                </div>
                    
                              </MDBCard>
                            </MDBCol>
                          </MDBRow>
                        </MDBContainer>


              </MDBCard>
            </MDBCol>
          </MDBRow>
        </MDBContainer>


      </div>
    )

}

export default Profile 