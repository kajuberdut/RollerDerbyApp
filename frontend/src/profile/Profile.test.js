import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import Profile from './Profile';

// * Passing as of 2/13/24
// npm test Profile.test.js
// must be in frontend directory 


afterEach(() => {
  cleanup()
})


describe("Profile tests ", () => {

  test('renders profile page with user name and number', () => {
    render(
      <BrowserRouter> 
          <Profile/>
      </BrowserRouter>
    );
    const userTitle = screen.getByText('testUser #01'); 
    expect(userTitle).toBeInTheDocument();
  });

  test('renders profile page test level', () => {
    render(
      <BrowserRouter> 
          <Profile/>
      </BrowserRouter>
    );
    const level = screen.getByText('level'); 
    expect(level).toBeInTheDocument();
    const userLevel = screen.getByText('A'); 
    expect(userLevel).toBeInTheDocument();
  });

  test('renders profile page test facebook name', () => {
    render(
      <BrowserRouter> 
          <Profile/>
      </BrowserRouter>
    );
    const facebookName = screen.getByText('You can find me on facebook: facebook test user'); 
    expect(facebookName).toBeInTheDocument();
  });

  test('renders profile page test about', () => {
    render(
      <BrowserRouter> 
          <Profile/>
      </BrowserRouter>
    );
    const about = screen.getByText('About'); 
    expect(about).toBeInTheDocument();
    const userAbout = screen.getByText('This is my story...'); 
    expect(userAbout).toBeInTheDocument();
  });

  test('renders profile page test associated leagues', () => {
    render(
      <BrowserRouter> 
          <Profile/>
      </BrowserRouter>
    );
    const associatedLeagues = screen.getByText('Associated Leagues'); 
    expect(associatedLeagues).toBeInTheDocument();
    const userAssociatedLeagues = screen.getByText('Testing League'); 
    expect(userAssociatedLeagues).toBeInTheDocument();
  });

});

describe("Snapshot Profile", () => {
  
  test('matches snapshot', function() {
    const { asFragment } = render(<Profile />);
    expect(asFragment()).toMatchSnapshot(); 
  })
  
  
});