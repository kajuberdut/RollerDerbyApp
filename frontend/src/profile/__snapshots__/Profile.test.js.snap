// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Profile matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
    style="background-color: transparent; padding: 100px 10% 10% 10%; margin-right: 0px;"
  >
    <div
      class="container"
      style="padding: 0px; margin: 0px;"
    >
      <div
        class="row"
        style="padding: 0px; margin: 0px;"
      >
        <div
          class="card"
          style="min-width: 612px; max-width: 1100px; min-height: 700px; margin-top: 50px; box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);"
        >
          <div
            class="rounded-top text-white d-flex flex-row"
            style="background-color: black; height: 300px; margin-top: 15px;"
          >
            <div
              class="ms-4 mt-5 d-flex flex-column"
              style="width: 250px;"
            >
              <img
                alt="Skater placeholder image"
                class="img-fluid mt-4 mb-2 img-thumbnail"
                src="/skater_02.svg"
                style="width: 250px; height: 330px; position: relative; z-index: 1; background-color: rgb(209, 210, 212); border: 4px solid white; box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);"
              />
            </div>
            <a
              href="/setup/profile"
            >
              <button
                class="btn btn-outline-dark"
                data-mdb-ripple-color="dark"
                style="height: 40px; background-color: rgb(209, 210, 212); position: absolute; right: 30px; top: 30px; font-size: 15px;"
                type="button"
              >
                Edit
              </button>
            </a>
            <a
              href="/profile/private"
            >
              <button
                class="btn btn-outline-dark"
                data-mdb-ripple-color="dark"
                style="height: 40px; background-color: rgb(209, 210, 212); position: absolute; right: 100px; top: 30px; font-size: 15px;"
                type="button"
              >
                Private Info
              </button>
            </a>
            <div
              class="ms-3"
              style="margin-top: 200px;"
            >
              <h4
                class=""
              >
                testUser #01
              </h4>
            </div>
          </div>
          <div
            class="p-4 text-black"
            style="background-color: rgb(248, 249, 250);"
          >
            <div
              class="d-flex justify-content-end text-center py-1"
              style="margin-top: 2px;"
            >
              <div
                style="margin-left: 600px;"
              >
                <div>
                  <p
                    class="card-text mb-1 h5"
                    style="margin-right: 80px;"
                  >
                    A
                  </p>
                  <p
                    class="card-text small text-muted mb-0"
                    style="margin-right: 80px;"
                  >
                    level
                  </p>
                </div>
              </div>
              <div>
                <p
                  class="card-text mb-1 h6"
                />
                <p
                  class="card-text small text-muted mb-0"
                  style="margin-right: 10px; margin-top: 7px;"
                >
                  positions
                </p>
              </div>
              <div
                class="px-3"
              >
                <p
                  class="card-text mb-1 h6"
                  style="margin-left: 30px;"
                />
                <p
                  class="card-text small text-muted mb-0"
                  style="margin-left: 30px; margin-top: 7px;"
                >
                  known rulesets
                </p>
              </div>
            </div>
          </div>
          <div
            class="card-body text-black p-4"
          >
            <div
              class="mb-5"
            >
              <p
                class="lead fw-normal mb-1"
              >
                About
              </p>
              <div
                class="p-4"
                style="background-color: rgb(248, 249, 250);"
              >
                <p
                  class="card-text font-italic mb-1"
                >
                  This is my story...
                </p>
              </div>
            </div>
            <div
              class="mb-5"
            >
              <p
                class="lead fw-normal mb-1"
              >
                Associated Leagues
              </p>
              <div
                class="p-4"
                style="background-color: rgb(248, 249, 250);"
              >
                <p
                  class="card-text font-italic mb-1"
                >
                  Testing League
                </p>
              </div>
            </div>
            <div
              class="mb-5"
            >
              <p
                class="lead fw-normal mb-1"
              >
                You can find me on facebook: facebook test user
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
